context/datapath.go

```diff
@@ -534,37 +476,19 @@ func (dpNode *DataPathNode) ActivateUpLinkPdr(smContext *SMContext, defQER *QER,
 
 	curULTunnel := dpNode.UpLinkTunnel
 	for name, ULPDR := range curULTunnel.PDR {
-		ULDestUPF := curULTunnel.DestEndPoint.UPF
 		ULPDR.QER = append(ULPDR.QER, defQER)
 
 		// Set Default precedence
 		if ULPDR.Precedence == 0 {
 			ULPDR.Precedence = defPrecedence
 		}
 
-		var iface *UPFInterfaceInfo
-		if dpNode.IsANUPF() {
-			iface = ULDestUPF.GetInterface(models.UpInterfaceType_N3, smContext.Dnn)
-		} else {
-			iface = ULDestUPF.GetInterface(models.UpInterfaceType_N9, smContext.Dnn)
+		ULPDR.PDI.SourceInterface = SourceInterface{InterfaceValue: SourceInterfaceAccess}
+		ULPDR.PDI.LocalFTeid = &FTEID{
+			Ch: true,
 		}
-
-		if upIP, err := iface.IP(smContext.SelectedPDUSessionType); err != nil {
-			logger.CtxLog.Errorf("activate UpLink PDR[%v] failed %v", name, err)
-			return err
-		} else {
-			ULPDR.PDI.SourceInterface = SourceInterface{InterfaceValue: SourceInterfaceAccess}
-			ULPDR.PDI.LocalFTeid = &FTEID{
-				V4:          true,
-				Ipv4Address: upIP,
-				Teid:        curULTunnel.TEID,
-			}
-
-			ULPDR.PDI.UEIPAddress = &ueIpAddr
-
-			ULPDR.PDI.NetworkInstance = util_3gpp.Dnn(smContext.Dnn)
-		}
-
+		ULPDR.PDI.UEIPAddress = &ueIpAddr
```

Comment:
How you are allocating UE address?

Reply:
The UE IP address allocation hasn't changed here, it is still done via `UeIPAllocator` (see producer/pdu_session.go` for more about this.